include/swift/Reflection/TypeLowering.h

```diff
@@ -155,6 +173,17 @@ class BuiltinTypeInfo : public TypeInfo {
     return Name;
   }
 
+  bool readExtraInhabitantIndex(remote::MemoryReader &reader,
+                                remote::RemoteAddress address,
+                                int *extraInhabitantIndex) const {
+    if (getNumExtraInhabitants() == 0) {
+      *extraInhabitantIndex = -1;
+      return true;
+    }
+    // XXX Has extra inhabitants, so it must be a pointer?
+    return reader.readHeapObjectExtraInhabitantIndex(address, extraInhabitantIndex);
```

Comment:
I'm not sure whether a `BuiltinTypeInfo` with extra inhabitants can ever be a function pointer or whether it's always just a regular pointer to heap data.  (There are different rules for managing XIs for function pointers.)